### ISSUE TITLE
TextField's decoration label should extend to right side even suffix icon is present

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#64427]
b/165825742

Right now it's cut to the same length as the available input width. Paste below code into dartpad.dev can see it visually:

```
import 'package:flutter/material.dart';

final Color darkBlue = Color.fromARGB(255, 18, 32, 47);

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(scaffoldBackgroundColor: darkBlue),
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: MyWidget(),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final text = 'Day of month to check';
    final width = 190.0;
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Container(
          padding: EdgeInsets.symmetric(horizontal: 12, vertical: 2),
          width: width,
          child: Text(text,
              textAlign: TextAlign.start,
              style: Theme.of(context).textTheme.caption),
        ),
        Container(
          width: width,
          child: TextField(
            decoration: InputDecoration(
              focusedBorder: OutlineInputBorder(
                borderSide: BorderSide(color: Colors.greenAccent, width: 1.0),
              ),
              enabledBorder: OutlineInputBorder(
                borderSide: BorderSide(color: Colors.grey, width: 1.0),
              ),
              suffixIcon: Icon(Icons.arrow_drop_down),
              floatingLabelBehavior: FloatingLabelBehavior.always,
              labelText: text,
            ),
          ),
        ),
      ],
    );
  }
}
```

We could potentially see the entire label text but now it's cut because it needs to be the same width as the input box.

**Comment by mehmetf** [comment]
This likely needs Material UX guidance.

**Comment by HansMuller** [comment]
CC @willlarche per https://github.com/flutter/flutter/issues/64427#issuecomment-678820364

**Comment by TahaTesser** [comment]
![Simulator Screen Shot - iPhone 11 - 2020-08-27 at 15 21 16](https://user-images.githubusercontent.com/48603081/91425625-032a1d80-e879-11ea-9dd8-0900f8ef6a90.png)



<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.93, on Mac OS X 10.15.6 19G2021,
    locale en-GB)
    • Flutter version 1.22.0-2.0.pre.93 at /Users/tahatesser/Code/flutter_master
    • Framework revision f35a8b7470 (5 hours ago), 2020-08-26 21:33:03 -0700
    • Engine revision 388193a67b
    • Dart version 2.10.0 (build 2.10.0-62.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • iPhone 11 (mobile) • A63D6F51-59F3-4A75-961F-3769709F3A92 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • Mac OS X 10.15.6 19G2021
    • Web Server (web)   • web-server                           • web-javascript
      • Flutter Tools
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 84.0.4147.135

• No issues found!

```

</details>

**Comment by willlarche** [comment]
Correct. It should extend to the right. But we do not have motion guidance on that transition. Do your best to extrapolate and let me know if you need an eye.

**Comment by YeungKC** [comment]
I'm interested in this issue, but extend to the right is it a target?

@mehmetf

**Comment by YeungKC** [comment]
![image](https://user-images.githubusercontent.com/11473691/94953598-66633b80-051a-11eb-9a1c-c92f144ef9a6.png)

If we simply change the width, which may not be very elegant in some extreme cases, I'm not sure if that's acceptable.
Or do we need the label to be higher than the icon? Or does the icon need to be reduced in height?

**Comment by willlarche** [comment]
The label doesn't need to be a tap target. So if the icon technically crosses the label, that's fine. It's transparent.

**Comment by YeungKC** [comment]
Yes, it is transparent in most cases.  I will pr in the near future.  Thank you for your reply.

**Comment by willlarche** [comment]
Very excited! Thank you for working on this PR.

**Comment by YeungKC** [comment]
Fixed by #68727
